src/view/actions/sendEvent.jsx

```diff
@@ -139,20 +134,20 @@ const getSettings = ({ values }) => {
 };
 
 const validationSchema = object().shape({
-  xdm: string().matches(
-    singleDataElementRegex,
-    "Please specify a data element"
-  ),
+  xdm: string().matches(singleDataElementRegex, {
+    excludeEmptyString: true,
```

Comment:
We initialize the form with an empty string for the `xdm` field. Upon validation, Formik would massage the data before passing it to validation such that the value being validated were not an empty string, which allowed this validation to pass. React Hook Form, on the other hand, doesn't attempt to massage the data and therefore passes an empty string to yup, which wouldn't normally pass validation because it doesn't match the regex. `excludeEmptyString` allows the validation to pass when the value is an empty string, which is what we want in this case.